rest-server/core/src/main/scala/com/actionml/core/store/DaoQuery.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright ActionML, LLC under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * ActionML licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.actionml.core.store
+
+
+case class DaoQuery(offset: Int = 0, limit: Int = Int.MaxValue, orderBy: Option[OrderBy] = None, filter: Seq[(String, Any)] = Seq.empty)
+
+object Ordering extends Enumeration {
+  type Ordering = Value
+  val asc = Value("asc")
+  val desc = Value("desc")
+}
+
+case class OrderBy(ordering: Ordering.Ordering, fieldNames: String*)
+
```

Comment:
can I ask for 100 events with name="buy" AND userId = "some-user" ordered by the event.date, descending? Can you show an example assuming event is:

```
case class Event(_id: ObjectId, userId: String, name: String, date: Datetime)
```

Use whatever datetime type you want. 

Also doesn't `_id`, `name`, `userId`, and `date` need to be indexed to make this query fast? How do we specify what is indexed? We can assume `_id` is because it is the default object id.

Reply:
Never mind, I can see how to encode the query but still have a question about indexes.